app/services/foreman/deprecation.rb

```diff
@@ -0,0 +1,19 @@
+module Foreman
+  class Deprecation
+    #aviliable options: controller, method, info, callstack
+    #deadline_version is the last version the method could be used
+    def self.deprecation_warning(foreman_version_deadline, options = {})
+      raise Foreman::Exception, N_("Invalid version format, please enter in x.y.z") unless foreman_version_deadline.to_s.match(/\A\d[.]\d+[.]\d\z/)
+      callstack = options[:callstack] || caller
+      log_msg = "You are using a deprecated behavior, #{foreman_version_deadline} is the last version it will be available for use."
+      log_msg += " Use #{options[:controller]}##{options[:method]} instead." if options[:controller] && options[:method]
+      log_msg += " #{options[:info]}." if options[:info]
+      ActiveSupport::Deprecation.warn(log_msg, callstack)
+      raise Foreman::Exception, N_("You are using a deprecated behavior, for more information check log file. %{callstack}") % {:callstack => ActiveSupport::Deprecation.send(:deprecation_caller_message, callstack)} if version_is_outdated?(foreman_version_deadline)
+    end
+
+    def self.version_is_outdated?(deadline_version)
+      Gem::Dependency.new('', "< #{SETTINGS[:version].notag}").match?('', deadline_version)
```

Comment:
Using "1.10.0" as the deadline version is going to cause exceptions on 1.10.1, a patch release, since 1.10.1 > 1.10.0.

You might want to change the versions passed in to be the *first* release that will actually remove them, rather than trying to guess what the last release to include them will be (since 1.10.0 wouldn't be it, it could be 1.10.5 or something), i.e. 1.11.0.